### DISCUSSION TITLE
Subpath/Relative path redirected to absolute path before logging in

### DISCUSSION THREAD

**Discussion by shuchenliu** [coder/code-server#2072] [Q&A] [answered]
- Web Browser: Safari/Chrome
- Local OS: MacOS
- Remote OS: CentOS 7
- Remote Architecture: `code-server` running as a service , `nginx` as reverse proxy
- `code-server --version`: 3.5.0

### Issue description:
Intended to run `code-server` behind a subpath/relative path of the host domain like `example.com/code`, however when visiting that url, browser got redirected to `example.com/login?to=%2F` instead of `example.com/code/login?to=%2F`. 

In reality, Nginx can be set thus `/login?to=%2F` is also mapped to code-server, which however defeats the purpose of setting a subpath in terms of the separation of services.

### Nginx config:

```nginx
server {
    listen 80;
    listen [::]:80;
    server_name <example.com>;

    location /code {
      proxy_pass http://localhost:8080/;
      proxy_set_header Host $host;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection upgrade;
      proxy_set_header Accept-Encoding gzip;
    }
}
```

### code-server `config.yaml`

```yaml
bind-addr: 127.0.0.1:8080
auth: password
password:<some password>
cert: false
```

### Failed attempts

Tried to set `base-url` in `config.yaml` however that option seems to have been removed ever since earlier releases and would result in an error when restarting `code-server`.

Any suggestion would be appreciated and thanks so much for your time.

**Comment by code-asher** [comment, marked as answer]
I think this is because there's no trailing slash on `/code`. code-server tries to redirect to a relative path (`./login`) which with `/code` ends up being `/login`.

One solution would be to use nginx to redirect `/code` to `/code/` and then proxy the latter to code-server.

**Reply by shuchenliu** [reply]
Gotcha, changing to `/code/` in Nginx config indeed solves the problem. Thanks a lot!
